pkg/nfs/nodeserver.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfs
+
+import (
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (ns *nfsDriver) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
+	return ns.nfsNodePublishVolume(ctx, req)
+}
+
+func (ns *nfsDriver) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublishVolumeRequest) (*csi.NodeUnpublishVolumeResponse, error) {
+	return ns.nfsNodeUnpublishVolume(ctx, req)
+}
+
+func (ns *nfsDriver) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstageVolumeRequest) (*csi.NodeUnstageVolumeResponse, error) {
+	return ns.nfsNodeUnstageVolume(ctx, req)
+}
+
+func (ns *nfsDriver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRequest) (*csi.NodeStageVolumeResponse, error) {
+	return ns.nfsNodeStageVolume(ctx, req)
+}
+
+func (ns *nfsDriver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
+	return &csi.NodeGetInfoResponse{
+		NodeId: ns.NodeID,
+		MaxVolumesPerNode: 100,
+	}, nil
+}
+
+func (ns *nfsDriver) NodeGetCapabilities(ctx context.Context, req *csi.NodeGetCapabilitiesRequest) (*csi.NodeGetCapabilitiesResponse, error) {
+	// TODO: check if SINGLE_NODE_MULTI_WRITER needs to be exported from here
```

Comment:
Node capability should export the following. NodePublish is always enabled as default capability.

-- SINGLE_NODE_MULTI_WRITER alone. As all exported volumes on windows are shared volumes enabling this flag is correct.

And Node Stage_Unstage is not needed anymore.

Please note that it differs from exported node capability on linux node by the following missing caps:

```
    caps := []csi.NodeServiceCapability_RPC_Type{
        // Getting volume stats for volume health monitoring
        csi.NodeServiceCapability_RPC_GET_VOLUME_STATS,
        // Indicates that the Node service can report volume conditions.
        csi.NodeServiceCapability_RPC_VOLUME_CONDITION,
    }
```